src/utils/wallet.js

```diff
@@ -18,16 +20,37 @@ export function WalletProvider({ children }) {
     'https://www.sollet.io',
   );
 
-  const wallet = useMemo(() => new Wallet(providerUrl, endpoint), [
+  let wallet = useMemo(() => new Wallet(providerUrl, endpoint), [
     providerUrl,
     endpoint,
   ]);
+  const [ezWallet] = useState(new EzWallet());
+  if (providerUrl === 'https://ezdefi.com/wallet') {
+    wallet = ezWallet;
+  }
 
   const [connected, setConnected] = useState(false);
+  const [cluster, setCluster] = useState(false);
 
   useEffect(() => {
     console.log('trying to connect');
+    if (wallet === ezWallet && !window.solana) {
```

Comment:
only 1 minor change is recommended: move the `window.solana` check into ezWallet adapter function:
```
if (wallet.isBrowserExtension && !wallet.isBrowserExtension())
```